src/java.base/share/classes/java/lang/foreign/MemorySession.java

```diff
@@ -228,56 +228,85 @@ default MemorySegment allocate(long bytesSize, long bytesAlignment) {
     }
 
     /**
-     * Creates a closeable confined memory session.
-     * @return a new closeable confined memory session.
+     * Creates and returns a new closeable, thread-confined memory session.
```

Comment:
This `Creates and returns` is correct technically, but I can't find much evidence for it in the rest of the JDK. It gives 109 matches in current code base, whereas `Creates` has over 7K. I would leave this as is.